addons/mrp/models/stock_move.py

```diff
@@ -290,3 +290,12 @@ def _compute_kit_quantities(self, product_id, kit_qty, kit_bom, filters):
             return min(qty_ratios) // 1
         else:
             return 0.0
+
+    def _search_picking_for_assignation(self):
+        res = super(StockMove, self)._search_picking_for_assignation()
+        for picking in res:
+            components = self.move_orig_ids.production_id.bom_id.bom_line_ids.mapped('product_id.id')
```

Comment:
I think this is a bit cleaner/less confusing:
```suggestion
            components = self.mapped('move_orig_ids.production_id.bom_id.bom_line_ids.product_id')
```